### PR TITLE
Enable external access to cli inside a spin instance

### PR DESCRIPTION
**Pull request by alvaro-shopify**
### WHY are these changes introduced?
part of  https://github.com/Shopify/shopify-cli-planning/issues/426
As part of the effort in [streamlining the first-party experience when creating apps](https://vault.shopify.io/gsd/projects/25765), we would like to make it as easy as possible for 1p developers to create apps on Spin.

Despite the environment used to run the `dev` command the CLI proceses must be accesible from the browser. Right now, the cli is not accessible when is run inside a spin instance. 

### WHAT is this pull request doing?
- Adds a `constellation` configuration to the CLI repo thus `1P users` can create a spin instance with the services needed to run CLI commands. Besides the port needed for the CLI to initiate the proceses for authentication, dev, etc, ... is opened in the `nginx` configuration. [Spin doc](https://vault.shopify.io/page/Preparing-repositories~dhbc579.md)

- In the same way that with the other cloud environments, the cli processes are configured using a public `spin domain url`  using the above configuration that allows external access to cli

- Modified `dev` [internal flowchart ](https://www.figma.com/file/PCukduZbTX4IWhC8mkGpHO/CLI3-Internals-(Copy)?node-id=0%3A1)

### How to test your changes?
- Checkout the branch used to create this PR `feature/add_external_access_to_cli_inside_spin`
- Create an spin instance using this command `spin up path/to/branch/.spin/constellation.yml -c cli.branch=feature/add_external_access_to_cli_inside_spin`
- Access to the spin instance `spin shell`
- Please make sure all the services are up and running. It's possible that some service is failing in case you receive `502` errors when opening the dashboard or using the apis through the CLI,
- A new development store should be created. In case the store is not listed once created, restart the partner service to see it immediately (it should be listed after waiting sometime. There is an issue with partners kafka consumers)
- Create a new app.
  - `cd cli`
  - `yarn install`
  - `yarn create-app --path path/to/cli-apps`
- Run `yarn shopify app dev`
- Finish the authentication process

- Preview is not working because the issue with the spin stores domains. Solved in this [PR](https://github.com/Shopify/cli/pull/553)
  
### Post-release steps0
If the users with older `partner` constellations updates the CLI version, the won't be able to run the CLI in spin, because in the new version, a `nginx` routing configuration is used instead of `ngrok` by default. The are two alternatives:
- Create a new spin instance that includes `partners` which will have the `nginx` configuration built-in
- Creating a manual tunnel inside spin and running the `dev` command with the flag `--tunell-url`

### Measuring impact

How do we know this change was effective? Please choose one:

- [ ] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes
- [x] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

File: .changeset/tidy-chefs-matter.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@shopify/app': major
+'@shopify/cli-kit': major
```
[the remark this answers is not in the document]
Comment (alvaro-shopify) [reply .changeset/tidy-chefs-matter.md:3]: 🤦

**Comment by alvaro-shopify** [comment]
`spin` configuration has been added for testing purposes. It should be removed prior to merge the PR

**Comment by pepicrft** [comment]
@alvaro-shopify I followed the 🎩 instructions, logged in with the `partner-dev` account, and ended up getting the following error in the terminal:

```
━━━━━━ Error ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

      The Partners GraphQL API responded unsuccessfully with the HTTP status 502 and errors:

      {}


━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
```
When sending the following query:

```
{
    organizations(first: 200) {
      nodes {
        id
        businessName
        website
        appsNext
      }
    }
  }
 ```

**Comment by isaacroldan** [comment]
That'd be a different constellation, many teams just want to work on apps or extensions and don't need the CLI repo for that. It's a different use case. We could discuss if a constellation for the CLI is necessary or not (considering this is a public project also)

**Comment by alvaro-shopify** [comment]
@pepicrft @isaacroldan I think we should have a specific private repo to define different constellation configurations to use the `cli` with spin. Those configurations should automate some of the manual configurations that should be made to `dev` different extensions types. For example, to preview the `checkout-ui` the user should be the owner of the store (or have enough rights) and populate some products. The idea is creating a repo like this https://github.com/Shopify/extensions-sandbox-app. By the way, that repo should be prepared to test extensions but unfortunately the scripts for configuring organization, store and app it's not working.
That new repo should also include the `nginx.conf` so the users could run the cli not only locally but also inside the spin instance

@pepicrft 
> I followed the 🎩 instructions, logged in with the partner-dev account, and ended up getting the following error in the terminal:
Sorry, the instructions were a bit outdated. I have added more information to them so there is no missing steps. Anyway, the `502` seems to be that the partners service was not started properly

**Review by isaacroldan — APPROVED** [review]

**Pull request merged**